File: semana_14_simulacro_parcial2/_Resolucion/VB_TPS_BugTracker/TPS_BugTracker/GUILayer/Bugs/frmNuevoBug.vb
Partial Public Class frmNuevoBug
    Inherits Form

    Private prioridadService As PrioridadService = New PrioridadService()
    Private productoService As ProductoService = New ProductoService()
    Private criticidadService As CriticidadService = New CriticidadService()

    Private bugService As BugService = New BugService()

    Public Sub New()
        InitializeComponent()
        GUIHelper.getHelper().llenarCombo(cboPrioridad, prioridadService.consultarPrioridades(), "nombre", "id_prioridad")
        GUIHelper.getHelper().llenarCombo(cboCriticidad, criticidadService.consultarCriticidades(), "nombre", "id_criticidad")
        GUIHelper.getHelper().llenarCombo(cboProducto, productoService.consultarProductos(), "nombre", "id_producto")
    End Sub

    Private Sub btn_aceptar_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btn_aceptar.Click
        Try

            If validarCampos() Then
                Dim nuevoBug As Bug = New Bug()
                nuevoBug.titulo = txtTitulo.Text
                nuevoBug.descripcion = txtDescripcion.Text
                nuevoBug.id_estado = 1
                nuevoBug.id_prioridad = CInt(cboPrioridad.SelectedValue)
                nuevoBug.id_criticidad = CInt(cboCriticidad.SelectedValue)
                nuevoBug.id_producto = CInt(cboProducto.SelectedValue)
                Dim historial As HistorialBug = New HistorialBug()
                historial.responsable = frmPrincipal.obtenerUsuarioLogin().id_usuario
                historial.fecha = DateTime.Now
                nuevoBug.addHistorial(historial)

                If bugService.crearBug(nuevoBug) Then
                    MessageBox.Show("El bug se creó correctamente", "Nuevo Bug", MessageBoxButtons.OK, MessageBoxIcon.Information)
                    Me.Close()
                Else
                    MessageBox.Show("No pudo crear el Bug", "Nuevo Bug", MessageBoxButtons.OK, MessageBoxIcon.Exclamation)
                End If
            End If

        Catch ex As Exception
            MessageBox.Show("Se produjo el siguiente error al crear un bug: " & ex.Message, "Error creando Bug", MessageBoxButtons.OK, MessageBoxIcon.[Error])
        End Try
    End Sub

    Private Function validarCampos() As Boolean
        If String.IsNullOrEmpty(txtTitulo.Text) Then
            MessageBox.Show("El campo Título es obligatorio.", "Validaciones", MessageBoxButtons.OK, MessageBoxIcon.Exclamation)
            Return False
        End If

        If cboProducto.SelectedIndex < 0 Then
            MessageBox.Show("El campo Producto es obligatorio.", "Validaciones", MessageBoxButtons.OK, MessageBoxIcon.Exclamation)
            Return False
        End If

        Return True
    End Function

    Private Sub btn_cancelar_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btn_cancelar.Click
        Me.Close()
    End Sub
End Class

File: semana_14_simulacro_parcial2/_Resolucion/C#_TPS_BugTracker/TPS_BugTracker/GUILayer/Bugs/frmNuevoBug.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using TPS_BugTracker.BusinessLayer.Entities;
using TPS_BugTracker.BusinessLayer.Services;
using TPS_BugTracker.DataLayer.DAOs;
using TPS_BugTracker.GUILayer.Helper;

namespace TPS_BugTracker.GUILayer.Bugs
{
    public partial class frmNuevoBug : Form
    {
        private PrioridadService prioridadService = new PrioridadService();
        private ProductoService productoService = new ProductoService();
        private CriticidadService criticidadService = new CriticidadService();

        private BugService bugService = new BugService();

        public frmNuevoBug()
        {
            InitializeComponent();

            GUIHelper.getHelper().llenarCombo(cboPrioridad, prioridadService.consultarPrioridades(), "nombre", "id_prioridad");

            GUIHelper.getHelper().llenarCombo(cboCriticidad, criticidadService.consultarCriticidades(), "nombre", "id_criticidad");

            GUIHelper.getHelper().llenarCombo(cboProducto, productoService.consultarProductos(), "nombre", "id_producto");
        }

        private void btn_aceptar_Click(object sender, EventArgs e)
        {
            try
            {
                if (validarCampos())
                {
                    //Creamos una instancia de Bug

                    Bug nuevoBug = new Bug();
                    nuevoBug.titulo = txtTitulo.Text;
                    nuevoBug.descripcion = txtDescripcion.Text;

                    nuevoBug.id_estado = 1; // Estado  "Nuevo"
                    nuevoBug.id_prioridad = (int)cboPrioridad.SelectedValue;
                    nuevoBug.id_criticidad = (int)cboCriticidad.SelectedValue;
                    nuevoBug.id_producto = (int)cboProducto.SelectedValue;

                    HistorialBug historial = new HistorialBug();
                    historial.responsable = frmPrincipal.obtenerUsuarioLogin().id_usuario;
                    historial.fecha = DateTime.Now;

                    nuevoBug.addHistorial(historial);

                    if (bugService.crearBug(nuevoBug))
                    {
                        MessageBox.Show("El bug se creó correctamente", "Nuevo Bug", MessageBoxButtons.OK, MessageBoxIcon.Information);
                        this.Close();
                    }
                    else
                    {
                        MessageBox.Show("No pudo crear el Bug", "Nuevo Bug", MessageBoxButtons.OK, MessageBoxIcon.Exclamation);
                    }
                }

            }
            catch (Exception ex)
            {
                MessageBox.Show("Se produjo el siguiente error al crear un bug: " + ex.Message, "Error creando Bug", MessageBoxButtons.OK, MessageBoxIcon.Error);

            }
        }

        private bool validarCampos()
        {
            if (string.IsNullOrEmpty(txtTitulo.Text))
            {
                MessageBox.Show("El campo Título es obligatorio.", "Validaciones", MessageBoxButtons.OK, MessageBoxIcon.Exclamation);

                return false;
            }

            if (cboProducto.SelectedIndex < 0)
            {
                MessageBox.Show("El campo Producto es obligatorio.", "Validaciones", MessageBoxButtons.OK, MessageBoxIcon.Exclamation);

                return false;
            }

            return true;
        }

        private void btn_cancelar_Click(object sender, EventArgs e)
        {
            this.Close();
        }



    }
}
